packages/flutter/lib/src/painting/border_radius.dart

```diff
@@ -398,12 +400,23 @@ class BorderRadius extends BorderRadiusGeometry {
     // Because the current radii could be negative, we must clamp them before
     // converting them to an RRect to be rendered, since negative radii on
     // RRects don't make sense.
+    //
+    // We also clamp the radii to be no larger than half the width or height of
+    // the rect, since radii larger than that don't make sense either.
+    // We only do this if the radii are circular, since non-circular radii
+    // can be larger than half the width or height of the rect.
```

Comment:
Please file a bug to figure out what to do for non-circular radii and add a TODO linking to it here. Ideally it should have details from what you've attempted so far and what works/doesn't work.